src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -349,6 +348,15 @@ private void SetCapacity(int capacity)
             _version++;
         }
 
+        // Increments the index wrapping it if necessary.
+        private void MoveNext(ref int index)
+        {
+            // It is tempting to use the remainder operator here but it is actually much slower 
+            // than a simple comparison and a rarely taken branch.   
+            int tmp = index + 1;
+            index = (tmp == _array.Length) ? 0 : tmp;
+        }
+
```

Comment:
Wouldn't this be better?

```c#
private void MoveNext(ref int index)
{
    index++;
    if (index == _array.Length)
    {
        index = 0;
    }
}
```

Reply:
Would the C# compiler automatically inline the MoveNext method in the release build? If so, then ref won't be slower than a local variable. Just wondering...